src/QueryStrategies/BatchQueryStrategies/batch_qs_base.jl

```diff
@@ -0,0 +1,130 @@
+div_measure computes diversity of all samples to one specific sample
+Iterative computation: only value for added sample needs to be computed, old_scores saves aggregated result
+"""
+function get_iterative_div_measure(name::Symbol)::Function
+    if (name == :AngleDiversity)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, candidate_indices::Vector{Int}, j::Int, old_scores::Vector{Float64}) -> begin
+            if model.data == data
+                # reuse model kernel matrix
+                K = SVDD.is_K_adjusted(model) ? model.K_adjusted : model.K
+            else
+                # compute a new kernel matrix because query data differs from model data
+                K = MLKernels.kernelmatrix(Val(:col), model.kernel_fct, data)
+            end
+            div_scores = [-abs(K[i,j]) / (sqrt(K[i,i]) * sqrt(K[j,j])) for i in candidate_indices]
+            if (length(old_scores) > 0)
+                div_scores = min.(div_scores, old_scores)
+            end
+            return div_scores
+        end
+    elseif (name == :EuclideanDistance)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, candidate_indices::Vector{Int}, j::Int, old_scores::Vector{Float64}) -> begin
+            # vector containing norms of columns
+            # use vec() as otherwise a row vector is returned as opposed to a column vector
+            div_scores = Distances.colwise(Distances.Euclidean(), data[:,candidate_indices], data[:,j])
+            if (length(old_scores) > 0)
+                div_scores = min.(div_scores, old_scores)
+            end
+            return div_scores
+        end
+    else
+        return throw(ArgumentError("Invalid diversity measure $(name) specified."))
+    end
+end
+
+"""
+div_measure computes diversity of the samples in a batch
+enumerative computation: compute diversity for the whole batch each time
+currently two measures are implemented:
+:AngleDiversity - Batch diversity is minimal angle between two batch samples in kernel space
+:EuclideanDistance - Batch diversity is minimal euclidean distance between two batch samples in feature space
+"""
+function get_enumerative_div_measure(name::Symbol)::Function
+    if (name == :AngleDiversity)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, batch::Vector{Int}) -> begin
+            if model.data == data
+                # reuse model kernel matrix
+                K = SVDD.is_K_adjusted(model) ? model.K_adjusted : model.K
+            else
+                # compute a new kernel matrix because query data differs from model data
+                K = MLKernels.kernelmatrix(Val(:col), model.kernel_fct, data)
+            end
+            min_div = Inf
+            batch_size = length(batch)
+            for i in 1:batch_size
+                for j in i+1:batch_size
+                    @inbounds div = -abs(K[i,j]) / (sqrt(K[i,i]) * sqrt(K[j,j]))
+                    min_div = min(div, min_div)
```

Comment:
This is more of a semantic comment: Why do we use `min` instead of average as diversity?